src/Lean/Elab/PreDefinition/WF/GuessLex.lean

```diff
@@ -364,13 +364,19 @@ def collectRecCalls (unaryPreDef : PreDefinition) (fixedParamPerms : FixedParamP
       RecCallWithContext.create (← getRef) caller callerParams callee calleeArgs
 
 /-- Is the expression a `<`-like comparison of `Nat` expressions -/
-def isNatCmp (e : Expr) : Option (Expr × Expr) :=
-  match_expr e with
-  | LT.lt α _ e₁ e₂ => if α.isConstOf ``Nat then some (e₁, e₂) else none
-  | LE.le α _ e₁ e₂ => if α.isConstOf ``Nat then some (e₁, e₂) else none
-  | GT.gt α _ e₁ e₂ => if α.isConstOf ``Nat then some (e₂, e₁) else none
-  | GE.ge α _ e₁ e₂ => if α.isConstOf ``Nat then some (e₂, e₁) else none
-  | _ => none
+def isNatCmp (e : Expr) : MetaM (Option (Expr × Expr)) := withReducible do
+  let (α, e₁, e₂) ←
+    match_expr e with
+    | LT.lt α _ e₁ e₂ => pure (α, e₁, e₂)
+    | LE.le α _ e₁ e₂ => pure (α, e₁, e₂)
+    | GT.gt α _ e₁ e₂ => pure (α, e₂, e₁)
+    | GE.ge α _ e₁ e₂ => pure (α, e₂, e₁)
+    | _ => return none
+
+  if ←isDefEq α (mkConst ``Nat) then
```

Comment:
Its not my natural style, thats why I didn't write it initially, but the predominant style is to always add parens around an embedded monadic operation, even if it's not syntactically required:
```suggestion
  if (← isDefEq α (mkConst ``Nat)) then
```